### PR TITLE
Schema

### LINKED ISSUE #131: Get table and view names + types
Given a SQL query

```
CREATE TABLE demographics (
    cc_num FLOAT64 NOT NULL,
);

CREATE TABLE transactions (
    trans_date_trans_time TIMESTAMP,
);

CREATE VIEW transactions_with_demographics as 
    SELECT
        transactions.cc_num,
    FROM
     ...
```

I'd like to get the following info in JSON: table names & type, view names, view type. e.g., something like that (actual schema is open to discussion):

```
{
  tables: [ {name: 'demographics', fields: [ { name: 'cc_num', type: 'float64', attributes: ['not null', 'unique'] }, ... ], ... }
  views: [ {name: 'transactions_with_demographics', fields: [{name: 'cc_num', type: float64, attributes: []}, ...]}, ...]
}
```

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
Output schema as JSON 
Fixes #131

### CODE REVIEW

**Comment by mbudiu-vmw** [comment]
Note: this uses the SQL type names, as the user will probably expect.

**Review by gz — APPROVED** [review]
I tried it and it worked

File: SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/statements/CreateRelationStatement.java [hunk 1]
```diff
@@ -114,4 +118,18 @@ public String toString() {
                 ", columns=" + this.columns +
                 '}';
     }
+
+    public JsonNode getDefinedObjectSchema() {
+        ObjectMapper mapper = new ObjectMapper();
+        ObjectNode result = mapper.createObjectNode();
+        result.put("name", this.tableName);
```
Comment (lalithsuresh) [inline SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/statements/CreateRelationStatement.java:125]: Will this later be usable to also get view definitions too? If so, it'd be good to specify whether something is a table or a view.

File: SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/DBSPCompiler.java [hunk 2]
```diff
@@ -94,11 +103,23 @@ enum InputSource {
...
         this.frontend = new CalciteCompiler(options);
         this.midend = new CalciteToDBSPCompiler(this.frontend, true, options, this);
         this.messages = new CompilerMessages(this);
         this.sources = new SourceFileContents();
         this.circuit = null;
+        if (options.ioOptions.emitJsonSchema != null) {
+            this.inputs = this.mapper.createArrayNode();
+            this.outputs = this.mapper.createArrayNode();
+            this.ios = this.mapper.createObjectNode();
+            this.ios.set("inputs", this.inputs);
```
Comment (lalithsuresh) [inline SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/visitors/DBSPCompiler.java:116]: Ah, I see now that this is where you handle inputs and outputs separately. Got it.

**Review by lalithsuresh — APPROVED** [review]

**Pull request merged**